bitnami/thanos/templates/receive/statefulset.yaml

```diff
@@ -92,7 +92,9 @@ spec:
             {{- else }}
             - --receive.local-endpoint=$(NAME).{{ include "common.names.fullname" . }}-receive-headless.$(NAMESPACE).svc.{{ .Values.clusterDomain }}:10901
             {{- end }}
+            {{- if eq .Values.receive.mode "standalone" }}
```

Comment:
So the Stateful Set should always have `receive.local-endpoint` set in either mode since in both modes the stateful set is what's storing the TSDB blocks. The condition for the local endpoint flag is essentially whether or not to create the distributor deployment. 

The flags in question behave like this within these two manifests:

* StatefulSet: Either has both `--receive.hashrings` and `--receive.local-endpoint` set in Standalone or just `--receive.local-endpoint` in dual-mode. 
* Distributor: Isn't deployed in Standalone or deployed with  just `--receive.hashrings`  in dual-mode